<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find aruco_ros)/custom_rosconsole.conf"/>
       
    <arg name="markerId"        default="19"/>
    <arg name="markerSize"      default="1"/>    <!-- in m -->
    <arg name="marker_frame"    default="aruco_map_align"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <node pkg="aruco_ros" type="single_mod" name="aruco_single_mod" output="screen">
        <remap from="/camera_info" to="/iris_landing_camera0/usb_cam/camera_info"/>
        <remap from="/image" to="/iris_landing_camera0/usb_cam/image_raw"/>
        
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="my_camera_frame"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

</launch>

